### ISSUE TITLE
Mocking chrono::Utc::now()

### ISSUE THREAD

**Issue by bbigras** [alexanderlinne/chronobreak#5]
Does chronobreak support chrono?

**Comment by alexanderlinne** [comment]
No, `chrono` is not supported yet. I unfortunately do not currently have any time to work on this project, so if you'd be interested to implement it, feel free to submit a PR.

**Comment by grapemix** [comment]
Thanks for this lib. I actually have the same problem. Is there any progress?

**Comment by git-staus** [comment]
I have the same use case.

I am unsure if I have the skill to deliver a viable PR, but what exactly is the reason `chrono` is not supported? Doesn't that crate ultimately rely on the time functionality in `std` as well?

**Comment by git-staus** [comment]
At first I thought this crate worked by somehow replacing the core clock functionality within `std`, which would mean that getting `chrono` to work should be trivial. Sadly is does not work like that:

```rust
 #[chronobreak] 
 use chrono::Utc; 
```

currently result in `` error[E0432]: unresolved import `chronobreak::mock::chrono` ``.

Actually implementing support for `chrono` requires duplicating its API surface in `chronobreak::mock::chrono`. This should be done with a newtype pattern on the unmocked `chrono` that simply passes the calls along to the real thing or mocks them with the functionality provided in `chronobreak::clock`. This latter module has (short) docstrings for if somebody wants to attempt to write such a PR, but I am not sure if I am capable of writing something on my own (or maybe I am simply too bad at Rust!)

For now my workaround will be to migrate from `chrono::Utc::now()` to something that works with `std::time::Instant::now()` and then mock that out with `chronobreak`.
